### PR TITLE
Properly escape codesign path with an apostrophe

### LINKED ISSUE #8703: Codesigning Problem when there is a Apostrophe in Path
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?
We have a project that has a Apostrophe and Space in the Target name. Using Xcode, trying to do a Build&Run fails because the codesign command does not properly escape the path to the signed framework.

## What did you expect to happen?
Build&Run should work.

## What happened instead?
The shell script Pods/Target Support Files/Pods-Quote's App/Quote's App-frameworks.sh is executed

You can see the error reports the path only until the apostrophe.

As a workaround I modified the code_sign_if_enabled() function as such:
```
# Signs a framework with the provided identity
code_sign_if_enabled() {
  if [ -n "${EXPANDED_CODE_SIGN_IDENTITY}" -a "${CODE_SIGNING_REQUIRED:-}" != "NO" -a "${CODE_SIGNING_ALLOWED}" != "NO" ]; then
    # Use the current code_sign_identitiy
    echo "Code Signing $1 with Identity ${EXPANDED_CODE_SIGN_IDENTITY_NAME}"
    local code_sign_cmd="/usr/bin/codesign --force --sign ${EXPANDED_CODE_SIGN_IDENTITY} ${OTHER_CODE_SIGN_FLAGS:-} --preserve-metadata=identifier,entitlements \"$1\""

    if [ "${COCOAPODS_PARALLEL_CODE_SIGN}" == "true" ]; then
      code_sign_cmd="$code_sign_cmd &"
    fi
    echo "$code_sign_cmd"
    eval "$code_sign_cmd"
  fi
}
```
With this modification the command works and the app builds.

### Stack

```
   CocoaPods : 1.5.3
        Ruby : ruby 2.3.7p456 (2018-03-28 revision 63024) [universal.x86_64-darwin18]
    RubyGems : 2.5.2.3
        Host : Mac OS X 10.14.2 (18C54)
       Xcode : 10.1 (10B61)
         Git : git version 2.17.2 (Apple Git-113)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ cefa0b01688a63ff44c71ecc749f4ed9158ec654
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.2
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.3.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'Quote\'s App' do
  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!
  pod 'SVProgressHUD'
  # Pods for Quote's App

end
```

### PR DESCRIPTION
**Pull request by SatbirTanda**
Fixes #8703 
Properly escapes path if it contains an apostrophe

### CODE REVIEW

**Comment by CocoaPodsBarista** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please include a CHANGELOG entry to credit yourself! <br />
You can find it at <a href="https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

Here's an example of your CHANGELOG entry:

```markdown
* Properly escape path  
  [SatbirTanda](https://github.com/SatbirTanda)
  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number)
```

*note*: There are two invisible spaces after the entry's text.


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by dnkoutso** [comment]
This change has happened in the past if I remember correctly but it broke something else. Maybe a target with spaces?

**Comment by dnkoutso** [comment]
I'll need to dig up some really old PRs though perhaps `git blame` those lines?

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Properly escape path
```
Comment (dnkoutso) [inline CHANGELOG.md]: this needs a better msg.

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,10 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Bug Fixes
 
+* Properly escape path  
+  [SatbirTanda](https://github.com/SatbirTanda)
+  [#issue_number](https://github.com/CocoaPods/CocoaPods/issues/issue_number/8703)
```
Comment (dnkoutso) [inline CHANGELOG.md]: replace #issue_number with the #8703

**Comment by dnkoutso** [comment]
see https://github.com/CocoaPods/CocoaPods/pull/6160 and https://github.com/CocoaPods/CocoaPods/pull/6120

We will need to add integration tests for these cases to ensure they all work. Spaces or apostrophes or dashes etc.

**Pull request closed without merging**